google-home-community.groovy

```diff
@@ -1489,38 +1491,64 @@ private deviceTraitPreferences_SensorState(deviceTrait) {
         deviceTrait.sensorTypes.each { sensorType ->
             // only display if the sensor has descriptive values
             if (GOOGLE_SENSOR_STATES[sensorType.key].descriptiveState) {
-                input(
-                    name: "${deviceTrait.name}.sensorTypes.${sensorType.key}.availableStates",
-                    title: "Google Home Available States for ${sensorType.key}",
-                    type: "enum",
-                    multiple: "true",
-                    required: "true",
-                    options: GOOGLE_SENSOR_STATES[sensorType.key].descriptiveState,
-                )
-                input(
-                    name: "${deviceTrait.name}.sensorTypes.${sensorType.key}.descriptiveAttribute",
-                    title: "Hubitat Descriptive State Attribute for ${sensorType.key}",
-                    type: "text",
-                    required: "true",
-                    defaultValue: GOOGLE_SENSOR_STATES[sensorType.key].descriptiveAttribute
-                )
+                // if the sensor does not have a numeric state, do not allow the user to disable the descriptive state
+                if (GOOGLE_SENSOR_STATES[sensorType.key].numericAttribute != "") {
+                    input(
+                        name: "${deviceTrait.name}.sensorTypes.${sensorType.key}.reportsDescriptiveState",
+                        title: "${sensorType.key} reports descriptive state",
+                        type: "bool",
+                        defaultValue: true,
+                        required: true,
+                        submitOnChange: true
+                    )
+                }
+                if (deviceTrait.sensorTypes[sensorType.key].reportsDescriptiveState) {
+                    input(
+                        name: "${deviceTrait.name}.sensorTypes.${sensorType.key}.availableStates",
+                        title: "Google Home Available States for ${sensorType.key}",
+                        type: "enum",
+                        multiple: "true",
+                        required: "false",
```

Comment:
I just noticed that you've used the strings `"true"` and `"false"` in a few places.  Both of those are truthy values.  Does this work?

Reply:
Yes, it works.  Not sure why I quoted them out as strings though.  I pushed a new patch to leave them as bare booleans.  Defaulted the other required back to true since they are required once the code gets to that spot (I was trying to deal with the non-init case that is now resolved).